python/valis/db/queries.py

```diff
@@ -690,8 +690,39 @@ def get_catalog_sources(sdss_id: int) -> peewee.ModelSelect:
     """
 
     s = vizdb.SDSSidFlat.select(vizdb.SDSSidFlat).where(vizdb.SDSSidFlat.sdss_id == sdss_id).alias('s')
-    return cat.Catalog.select(cat.Catalog, starfields(s)).\
-        join(s, on=(s.c.catalogid == cat.Catalog.catalogid)).order_by(cat.Catalog.version.desc())
+    return cat.Catalog.select(cat.Catalog, cat.SDSS_ID_To_Catalog, starfields(s)).\
+        join(s, on=(s.c.catalogid == cat.Catalog.catalogid)).\
+        join(cat.SDSS_ID_To_Catalog, on=(s.c.catalogid == cat.SDSS_ID_To_Catalog.catalogid)).\
+        order_by(cat.Catalog.version.desc())
+
```

Comment:
what made you change to integrating this into the `get_catalog_sources` query?  And are we ok with this being flattened?

Reply:
I think it probably saves a bit of overhead by doing a single query instead of two per `/target/catalogs` request. The end result is not flattened. In `get_catalogs` in the definition of the route I get the flattened result of that query and nest the parent catalogues inside a dictionary. I'm happy with that implementation but also to get back to the previous one.